src/flowchem/devices/vapourtec/r2_components_control.py

```diff
@@ -204,18 +205,19 @@ def _change_connections(self, raw_position, reverse: bool = False) -> str:
             translated = raw_position
         else:
             translated = raw_position
-        return translated
+        return translated # Check the return (translated is a int, right?)
 
     async def get_position(self) -> list[list]:
         """Get current valve position."""
+        # Check the return (position is a int, right?)
         position = await self.hw_device.get_valve_position(self.valve_code)
         return (self._positions[int(self._change_connections(position, reverse=True))])
 
     async def set_position(self, positions_to_connect):
         """Move valve to position, which connects named ports. For example, [[5,0]] or [[2,3]]"""
         positions_to_connect_l = json.loads(positions_to_connect)
         position_to_connect = tuple(tuple(inner_list) for inner_list in positions_to_connect_l)
-        target_pos = self._connect_positions(position_to_connect)
+        target_pos = self._connect_positions(position_to_connect) # Check the return - expected a tuple[tuple]
```

Comment:
check the return/remove that highliught

Reply:
I will check it. But not in this branch.